packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -892,7 +892,7 @@ export function commitHydratedSuspenseInstance(
 export function shouldDeleteUnhydratedTailInstances(
   parentType: string,
 ): boolean {
-  return parentType !== 'head' || parentType !== 'body';
+  return parentType !== 'head' && parentType !== 'body';
```

Comment:
This case is most for this case of hydrating the whole doc which doesn't have other test coverage. Made a typo in the refactor before.

Reply:
@acdlite This affects the reconciler.